mino/minogrpc/server.go

```diff
@@ -666,12 +672,41 @@ func (mgr *connManager) Acquire(to mino.Address) (grpc.ClientConnInterface, erro
 	}
 
 	// Connecting using TLS and the distant server certificate as the root.
-	conn, err = grpc.Dial(netAddr.GetDialAddress(),
+	addr := netAddr.GetDialAddress()
```

Comment:
This function is getting humongous. 
Consider splitting lines 645-667 into a separate function,